src/Ocelot.Provider.Polly/CircuitBreaker.cs

```diff
@@ -1,12 +1,23 @@
-namespace Ocelot.Provider.Polly
+namespace Ocelot.Provider.Polly;
+
+public class CircuitBreaker<TResult>
+    where TResult : class
 {
-    public class CircuitBreaker
+    /// <summary>
+    /// Initializes a new instance of the <see cref="CircuitBreaker{TResult}"/> class.
+    /// We expect at least one policy to be passed in, default can't be null.
+    /// </summary>
+    /// <param name="policies">The policies with at least a <see cref="Policy.Timeout(int)"/> policy.</param>
+    public CircuitBreaker(params IAsyncPolicy<TResult>[] policies)
     {
-        public CircuitBreaker(params IAsyncPolicy[] policies)
+        var allPolicies = policies.Where(p => p != null).ToArray();
+        CircuitBreakerAsyncPolicy = allPolicies.First();
```

Comment:
Are you sure that 1st array item will be `CircuitBreakerAsyncPolicy` matching the type?
Maybe you need a predicate by type here...

Reply:
We could just call it OcelotAsyncPolicy? The policies are wrapped together here.

Reply:
@raman-m Ok, since the timeout is part of the policies, it can't be named CircuitBreaker. I changed the names to be more precise: public PollyPolicyWrapper and then AsyncPollyPolicy for the wrapped policies.